webapp/components/AsyncComponent.jsx

```diff
@@ -38,20 +38,20 @@ export default class AsyncComponent extends Component {
     };
   }
 
-  componentWillMount() {
+  componentDidMount() {
     this.api = new Client();
     this.reloadData();
   }
 
-  componentWillReceiveProps(nextProps, nextContext) {
-    let isRouterInContext = !!nextContext.router;
-    let isLocationInProps = nextProps.location !== undefined;
+  componentDidUpdate(prevProps) {
```

Comment:
50% chance this breaks behavior, but given the router object is likely the same, the behavior might be the same